src/calibration.cpp

```diff
@@ -760,74 +764,65 @@ void Calibration::dacOutputDC(struct iio_device *dac,
 	if (!(*buffer)) {
 		qDebug(CAT_CALIBRATION) << "Could not create buffer for: "  <<
 			 iio_device_get_name(dac);
-		return;
+		return false;
 	}
 
 	std::fill_n(data, size, value);
 	iio_channel_write(channel, *buffer, data, size * sizeof(data[0]));
 
+
 	iio_buffer_push(*buffer);
+	return true;
 }
 
-void Calibration::dacAOutputDCVolts(int16_t dac_a_volts)
+bool Calibration::dacAOutputDCVolts(int16_t dac_a_volts)
 {
 	int dac_a_raw;
 	setCalibrationMode(NONE);
 	setChannelEnableState(m_dac_a_channel, true);
 	iio_device_attr_write_bool(m_m2k_dac_a, "dma_sync", true);
 	dac_a_raw = (dac_a_volts / m_dac_a_ch_vlsb);
-	dacAOutputDC(dac_a_raw);
+	bool ok = dacAOutputDC(dac_a_raw);
 
 	iio_device_attr_write_bool(m_m2k_dac_a, "dma_sync", false);
 
 	iio_channel_attr_write_bool(m_dac_a_fabric, "powerdown", false);
+	return ok;
 }
 
-void Calibration::dacBOutputDCVolts(int16_t dac_b_volts)
+bool Calibration::dacBOutputDCVolts(int16_t dac_b_volts)
 {
 	int dac_b_raw;
 	setCalibrationMode(NONE);
 	setChannelEnableState(m_dac_b_channel, true);
 	iio_device_attr_write_bool(m_m2k_dac_b, "dma_sync", true);
 	dac_b_raw = (dac_b_volts / m_dac_b_ch_vlsb);
-	dacBOutputDC(dac_b_raw);
+	bool ok = dacBOutputDC(dac_b_raw);
 
 	iio_device_attr_write_bool(m_m2k_dac_b, "dma_sync", false);
 
 	iio_channel_attr_write_bool(m_dac_b_fabric, "powerdown", false);
+	return ok;
 }
 
-void Calibration::dacAOutputDC(int16_t value)
+bool Calibration::dacAOutputDC(int16_t value)
 {
-	dacOutputDC(m_m2k_dac_a, m_dac_a_channel, &m_dac_a_buffer, value);
+	return dacOutputDC(m_m2k_dac_a, m_dac_a_channel, &m_dac_a_buffer, value);
 }
 
 void Calibration::dacOutputStop()
 {
-	if (m_dac_a_buffer) {
-		iio_buffer_cancel(m_dac_a_buffer);
-		iio_buffer_destroy(m_dac_a_buffer);
-		m_dac_a_buffer = NULL;
-	}
 
-	if (m_dac_b_buffer) {
-		iio_buffer_cancel(m_dac_b_buffer);
```

Comment:
iio_buffer_cancel in no longer called in turnDACOutputOff(). It this ok?

Reply:
You are correct, it should be called. I'll add this.